src/Emit.ts

```diff
@@ -11,18 +12,15 @@ export interface OutputFile {
 
 let lualibContent: string;
 export function emitTranspiledFiles(
+    program: ts.Program,
     options: CompilerOptions,
     transpiledFiles: TranspiledFile[],
     emitHost: EmitHost = ts.sys
 ): OutputFile[] {
-    let { rootDir, outDir, luaLibImport } = options;
+    let { outDir, luaLibImport } = options;
 
-    const configFileName = options.configFilePath as string | undefined;
-    // TODO: Use getCommonSourceDirectory
-    const baseDir = configFileName ? path.dirname(configFileName) : process.cwd();
-
-    rootDir = rootDir || baseDir;
-    outDir = outDir ? path.resolve(baseDir, outDir) : rootDir;
+    const rootDir = getProjectRootDir(program);
+    outDir = outDir ? path.resolve(rootDir, outDir) : rootDir;
```

Comment:
`outDir` should be resolved from config (though we probably don't need to resolve it at all, TypeScript resolves path during parsing)